Client/src/Network/NetworkClient.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type Client
+** File description:
+** NetworkClient
+*/
+
+#include <array>
+#include <thread>
+#include <string>
+#include <iomanip>
+#include <asio.hpp>
+#include <iostream>
+
+#include "../../../Packet/NetworkPacket.hpp"
+
+/**
+ * @namespace Network
+ * @brief Network-related classes and functions
+ */
+namespace Network {
+
+    /**
+     * @brief Client class to Network (Connection to server, receive data and send data)
+     */
+    class Client;
+};
```

Comment:
Why is the client class initially initialized and then completed?

Reply:
This is fine.

Reply:
I think this is more clear to have the definition of the namespace and after the initialisation because if we have a lot of class that going to be ugly

Reply:
I just asked the question to understand :)